### PR TITLE
Added tests for code coverage

### PR DESCRIPTION
**Pull request by makdharma**

### CODE REVIEW

File: src/objective-c/tests/GRPCClientTests.m [hunk 1]
```diff
@@ -41,11 +41,11 @@
 #import <RemoteTest/Messages.pbobjc.h>
 #import <RxLibrary/GRXWriteable.h>
 #import <RxLibrary/GRXWriter+Immediate.h>
-
+#import <GRPCClient/GRPCCall+ChannelArg.h>
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: nit: new line between imports and code

File: src/objective-c/tests/GRPCClientTests.m [hunk 2]
```diff
@@ -127,8 +127,8 @@ - (void)testConnectionToRemoteServer {
     XCTFail(@"Received unexpected response: %@", value);
   } completionHandler:^(NSError *errorOrNil) {
     XCTAssertNotNil(errorOrNil, @"Finished without error!");
-    // TODO(jcanizales): The server should return code 12 UNIMPLEMENTED, not 5 NOT FOUND.
-    XCTAssertEqual(errorOrNil.code, 5, @"Finished with unexpected error: %@", errorOrNil);
+    //
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: nit: remove line with empty comment

File: src/objective-c/tests/GRPCClientTests.m [hunk 3]
```diff
@@ -200,7 +200,7 @@ - (void)testMetadata {
   request.fillOauthScope = YES;
   GRXWriter *requestsWriter = [GRXWriter writerWithValue:[request data]];
 
-  GRPCCall *call = [[GRPCCall alloc] initWithHost:kHostAddress
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: Is `kHostAddress` still used elsewhere in the file?

File: src/objective-c/tests/InteropTestsLocalCleartext.m [hunk 4]
```diff
@@ -49,7 +49,7 @@ + (NSString *)host {
 
 - (void)setUp {
   // Register test server as non-SSL.
-  [GRPCCall useInsecureConnectionsForHost:kLocalCleartextHost];
+  [GRPCCall useInsecureConnectionsForHost:nil];
```
Comment (jcanizales) [inline src/objective-c/tests/InteropTestsLocalCleartext.m]: :D

File: src/objective-c/tests/InteropTestsLocalSSL.m [hunk 5]
```diff
@@ -57,4 +57,15 @@ - (void)setUp {
   [super setUp];
 }
 
+- (void)testExceptions {
```
Comment (jcanizales) [inline src/objective-c/tests/InteropTestsLocalSSL.m]: This is an odd place to put this test. Maybe create a GRPCCallUnitTests file and put there all tests that don't require a server?

File: src/objective-c/tests/RxLibraryUnitTests.m [hunk 6]
```diff
@@ -137,4 +137,27 @@ - (void)testBufferedPipePropagatesError {
...
+  // then start the writeable
+  [pipe startWithWriteable:writeable];
+
+  // Then:
+  XCTAssertEqual(handler.timesCalled, 1);
+  XCTAssertEqualObjects(handler.value, anyValue);
+  XCTAssertEqualObjects(handler.errorOrNil, nil);
+}
+
+- (void)testBufferedPipeSetState {
```
Comment (jcanizales) [inline src/objective-c/tests/RxLibraryUnitTests.m]: :)

File: src/objective-c/tests/GRPCClientTests.m (hunk 3 above)
Comment (makdharma) [inline src/objective-c/tests/GRPCClientTests.m]: Yes, by other tests.

File: src/objective-c/tests/InteropTestsLocalSSL.m (hunk 5 above)
Comment (makdharma) [inline src/objective-c/tests/InteropTestsLocalSSL.m]: I'll add a todo to create a new file. For now I'd like to keep the file structure so we can reuse the coverage information.

File: src/objective-c/tests/InteropTestsLocalCleartext.m (hunk 4 above)
Comment (jcanizales) [inline src/objective-c/tests/InteropTestsLocalCleartext.m]: This needs to be reverted.

File: src/objective-c/tests/GRPCClientTests.m [hunk 7]
```diff
@@ -257,4 +258,40 @@ - (void)testResponseMetadataKVO {
   [self waitForExpectationsWithTimeout:8 handler:nil];
 }
 
+// todo(makaradd): Move to a different file that contains only unit tests
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: nit: TODO in caps

File: src/objective-c/tests/GRPCClientTests.m [hunk 8]
```diff
@@ -200,7 +201,7 @@ - (void)testMetadata {
   request.fillOauthScope = YES;
   GRXWriter *requestsWriter = [GRXWriter writerWithValue:[request data]];
 
-  GRPCCall *call = [[GRPCCall alloc] initWithHost:kHostAddress
+  GRPCCall *call = [[GRPCCall alloc] initWithHost:kRemoteSSLHost
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: So is this test enabled now? If so, please remove the TODO on top of the method.

File: src/objective-c/tests/RxLibraryUnitTests.m (hunk 6 above)
Comment (jcanizales) [inline src/objective-c/tests/RxLibraryUnitTests.m]: This test is empty.

File: src/objective-c/tests/GRPCClientTests.m [hunk 9]
```diff
@@ -41,10 +41,12 @@
 #import <RemoteTest/Messages.pbobjc.h>
 #import <RxLibrary/GRXWriteable.h>
 #import <RxLibrary/GRXWriter+Immediate.h>
+#import <GRPCClient/GRPCCall+ChannelArg.h>
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: Last thing: imports should be ordered alphabetically within their groups.

**Comment by jcanizales** [comment]
Python Mac failures can't be related to this, so I'm merging.

**Pull request merged**